core/frontend/src/components/vehiclesetup/configuration/compass/CalibrationQualityIndicator.vue

```diff
@@ -0,0 +1,81 @@
+<template>
+  <div ref="bar" class="color-bar">
+    <span
+      ref="display"
+      :style="{ left: computed_left + 'px', 'background-color': computed_color }"
+      class="signal-quality"
+    >
+      {{ quality.toFixed(0) }}
+    </span>
+  </div>
+</template>
+
+<script lang="ts">
+
+export default {
+  name: 'CalibrationQualityIndicator',
+  props: {
+    quality: {
+      type: Number,
+      required: true,
+    },
+
+  },
+  data: () => ({
+    bar_width: 0,
+    display_width: 0,
+  }),
+  computed: {
+    computed_color() {
```

Comment:
We should use something like this:

https://www.npmjs.com/package/javascript-color-gradient